app/app/static/custom/ourscript.js

```diff
@@ -4,85 +4,215 @@
+	<td class="td-actions text-right">
+	<button type="button" rel="tooltip" class="btn btn-success" onclick="finishRow('${element.id}','${element.status}')(this)"><i class="material-icons">how_to_reg</i></button>
+	</tr>`).join("")
+}
 function showAddToQueue() {
 	$('#addToQueue').css('display', 'block');
 }
 
-function hideAddToQueue() {
+function hideAddToQueueForm() {
 	document.getElementById('addToQueueForm').reset();
 	$('#addToQueue').css('display', 'none');
 }
 
 $('#addToQueueForm').submit(function (e) {
+	// This is the function that executes on the submission of the Student Data
 	e.preventDefault();
 
-	let queue_id;
-	let name = document.getElementById('studentName').value;
-	let id = document.getElementById('studentNumber').value;
-	let unit = document.getElementById('unitCode').value;
-	let team = document.getElementById('team').value;
-	let enquiry = document.getElementById('enquiryType').value;
+	const studentName = document.getElementById('studentName').value;
+	const studentNumber = document.getElementById('studentNumber').value;
+	const unitCode = document.getElementById('unitCode').value;
+	const queue = document.getElementById('queue').value;
+	const enquiry = document.getElementById('enquiry').value;
 
-	
-	fetch("add_entry", {
-        method: "POST",
-        headers: {
-            studentName: name,
-            studentNumber: id,
-            unitCode: unit,
-            enquiry: enquiry,
-            queue: team
-        }
-    })
-
-	const table = $(`#${team == 'STUDYSmarter' ? 'SS' : 'lib'}QueueTable tbody`);
-	table.append(
-			`<tr id="${id}" class="initialTime">
-			<td>${name}</td>
-			<td>${id}</td>
-			<td>${unit}</td>
-			<td class="text-right">${enquiry}</td>
-			<td class="text-right"><label id="minutes${id}">00</label><label id="colon">:</label><label id="seconds${id}">00</label></td>
-			<td class="td-actions text-right">
-			<button type="button" rel="tooltip" class="btn btn-success" onclick="addSessionToTeam('${team}',${id})(this)"><i class="material-icons">how_to_reg</i></button>
-			<button type="button" rel="tooltip" class="btn btn-danger" onclick="deleteRow(this)"><i class="material-icons">close</i></button></td>
-			</tr>`
-	);
-
-	hideAddToQueue();
-	timers[id] = 0;
-	timerIntervals[id] = setInterval(setTime, 1000, id);
+	// Adds to state and rerender
+	addToQueueList({
+		studentName,
+		studentNumber,
+		unitCode,
+		queue,
+		enquiry
+	})
+
+	hideAddToQueueForm();
 });
 
-function deleteRow(x) {
-	$(x).parents('tr').remove();
+function deleteRow(id, status) {
+	const closureFunction = (currentElement) => {
+		console.log(currentElement)	
+		status = "Ended"
+		terminateSession({
+			id,
+			status
+		})
+}
+return closureFunction
 }
 
-function addSessionToTeam(team, id){
+function finishRow(id, status) {
+	const closureFunction = (currentElement) => {
+		console.log(currentElement)	
+		status = "Completed"
```

Comment:
What's the difference between this variable status and the function parameter status?

Reply:
totally not related, the function parameter should only be 'id'. Fixed!